### PR TITLE
Extend UTF16 encoding support for helidon-enforcer plugin (copyright rule).

### LINKED ISSUE #908: helidon-enforcer plugin (copyright rule) fails for UTF16 encoding
In EclipseLink project are for some testing purposes some files stored in UTF-16 encoding. In this case helidon-enforcer plugin will fail during copyright check.

### PR DESCRIPTION
**Pull request by rfelcman**
Fix plus unit test.
Fixes #908

### CODE REVIEW

**Comment by romain-grecourt** [comment]
Hi, let me know if you plan on signing the contributor agreement.

**Comment by rfelcman** [comment]
> Hi, let me know if you plan on signing the contributor agreement.

I really want, but as a Oracle employee I should be member of https://github.com/oracle , but It's unclear to me how to do it.
https://github.com/helidon-io/helidon-build-tools/pull/909/checks?check_run_id=12187533848

**Comment by rfelcman** [comment]
> Hi, let me know if you plan on signing the contributor agreement.

Request to be member [Oracle GitHub organization](https://github.com/oracle) according [oracle-contributor-agreement](https://github.com/apps/oracle-contributor-agreement) is in progress.

File: maven-plugins/enforcer-maven-plugin/src/main/java/io/helidon/build/maven/enforcer/copyright/ValidatorBase.java [hunk 1]
```diff
@@ -67,8 +67,14 @@ public void validate(FileRequest file, Path path) {
 
         try (BufferedReader br = Files.newBufferedReader(path, StandardCharsets.UTF_8)) {
             copyrightComment = readComment(file, br);
-        } catch (IOException e) {
-            throw new EnforcerException("Failed to read file " + path, e);
+        } catch (IOException e1) {
+            //Fallback to cover at least UTF-16 encoding
+            //Encoding autodetection should be better, but depends on 3rd party library
+            try (BufferedReader br = Files.newBufferedReader(path, StandardCharsets.UTF_16)) {
```
Comment (romain-grecourt) [inline maven-plugins/enforcer-maven-plugin/src/main/java/io/helidon/build/maven/enforcer/copyright/ValidatorBase.java:73]: Can you fallback on UTF-16 only if e1 is an instance of `CharacterCodingException` ?

Comment (rfelcman) [reply maven-plugins/enforcer-maven-plugin/src/main/java/io/helidon/build/maven/enforcer/copyright/ValidatorBase.java:74]: OK, changed, but `IOException` has to be handled too due `Files.newBufferedReader` and `readComment(`.

**Review by romain-grecourt — APPROVED** [review]

**Comment by romain-grecourt** [comment]
Inactive, closing.

**Pull request closed without merging**